src/adapter.js

```diff
@@ -0,0 +1,112 @@
+// API Blueprint serializer for Fury.js
+
+import mson from './mson';
+import nunjucks from 'nunjucks';
+
+const env = nunjucks.configure(__dirname, {
+  autoescape: false,
+});
+
+/*
+ * Indent every line except the first by a number of spaces.
+ *
+ * e.g. indent('foo\nbar\nbaz', 4)
+ *      => 'foo\n    bar\n    baz'
+ */
+function indent(input, spaces) {
+  let pre = '';
+  const lines = [];
+
+  for (let _ = 0; _ < spaces; _++) {
+    pre += ' ';
+  }
+
+  for (const line of input.split('\n')) {
+    lines.push(pre + line);
+  }
+
+  return lines.join('\n').trim();
+}
+
+/*
+ * Given a message payload, return true if it only has a body defined and
+ * optionally has a content-type and nothing else. This lets us know when to
+ * use a shorthand syntax in the template.
+ */
+function bodyOnly(payload) {
+  let headers;
+
+  // First, we need to filter out the content-type header. This is handled
+  // outside of the payload (e.g. `+ Response (application/json)`)
+  if (payload.headers) {
+    headers = payload.headers.exclude('Content-Type');
+  }
+
+  return payload.messageBody && !(headers.length || payload.dataStructure ||
+                                  payload.messageBodySchema);
+}
+
+/*
+ * Detect when it is okay to use a resource shorthand, i.e. skip the resource
+ * header and just do a single action with a URI.
+ */
+function resourceShorthand(resource) {
+  return (!resource.title && !resource.description &&
+          resource.transitions.length === 1 &&
+          resource.transitions.get(0).computedHref);
+}
+
+/*
+ * Attempt to print out a pretty version of some input. Currently supported
+ * inputs include:
+ *
+ * - JSON
+ *
+ */
+function pretty(input) {
+  let prettified;
+
+  try {
+    prettified = JSON.stringify(JSON.parse(input), null, 2);
+  } catch (err) {
+    prettified = input;
+  }
+
+  return prettified;
+}
+
+/*
+ * Return all child elements with the element type of `copy` in a plain
+ * old js array.
+ */
+function getCopy(element) {
+  return element.children((item) => item.element === 'copy').content;
+}
+
+env.addFilter('mson', mson);
+env.addFilter('indent', indent);
+env.addFilter('bodyOnly', bodyOnly);
+env.addFilter('resourceShorthand', resourceShorthand);
+env.addFilter('pretty', pretty);
+env.addFilter('getCopy', getCopy);
+
+export const name = 'api-blueprint-serializer';
+export const mediaTypes = [
+  'text/vnd.apiblueprint',
```

Comment:
Shouldn't this be `text/vnd.apiblueprint+markdown`?

ref https://github.com/apiaryio/apiblueprintorg/blob/master/apiary.apib#L15